pkg/scheduler/factory/factory.go

```diff
@@ -653,6 +653,17 @@ func (c *configFactory) addNodeToCache(obj interface{}) {
 		return
 	}
 
+	for _, cond := range node.Status.Conditions {
```

Comment:
It would be nice if a helper existed for getting the value of a particular condition. Not really in scope for this PR though.